cmds/core/uptime/uptime_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+type UptimeOutput struct {
+	uptime *time.Time
+	err    string
+}
+
+type LoadAvgOutput struct {
+	loadavg string
+	err     string
+}
+
+func TestUptimeGoodData(t *testing.T) {
+	time := time.Date(0001, 1, 15, 5, 35, 49, 0, time.UTC)
+	want := UptimeOutput{&time, ""}
+	uptime, err := uptime("1229749 1422244")
+	var got UptimeOutput
+	if err == nil {
+		got = UptimeOutput{uptime, ""}
+	} else {
+		got = UptimeOutput{uptime, err.Error()}
+	}
+	if !got.uptime.Equal(*want.uptime) {
+		t.Errorf("the uptime got is not correct:want %v , got %v", want.uptime, got.uptime)
+	}
+	if got.err != want.err {
+		t.Errorf("expected no error from a good proc/uptime file but got this error:%v", got.err)
+	}
+}
+
+func TestUptimeBadData(t *testing.T) {
+	time, err := uptime("string")
+	var got UptimeOutput
+	if err == nil {
+		got = UptimeOutput{time, ""}
+	} else {
+		got = UptimeOutput{time, err.Error()}
+	}
+	want := UptimeOutput{nil, "Error time: invalid duration strings"}
+	if got.uptime != want.uptime {
```

Comment:
Don't compare the pointers, compare the actual values. I think you need to do *got.uptime != *want.uptime

Reply:
Missed the fact that uptime is nil here. I'm going to recommend you move to table driven tests so you can better understand what I'm trying to drive at.